engine/setup/src/include/general.cpp

```diff
@@ -79,21 +87,28 @@ char *split_words(char *string, int max_words) {
 
 // Better yet, use glib
 
-char *ptr_copy(char *string) {
+char *ptr_copy(char *string) 
+{
 	char *alloc;
-	alloc = (char *)malloc(strlen(string)+1);
-	if (alloc == 0) { fprintf(stderr, "Out of memory\n"); exit(-1); }
-	strncpy(alloc, string, strlen(string));
+	alloc = (char *)malloc(strlen(string) + 1);
+	if (alloc == 0) 
+        {
+            fprintf(stderr, "Out of memory\n"); 
+            exit(-1);
+        }
+	alloc = string;
```

Comment:
This isn't quite correct either, since it will simply end up returning the original string, rather than a copy of the string like it's supposed to.

This function was bugging me, so I spent some time on it today, and came up with a rather over-engineered solution. LOL.

Here it is. Function `ptr_copy` becomes:

```C++
char *ptr_copy(const char *string)
{
#if _XOPEN_SOURCE >= 500 || _POSIX_C_SOURCE >= 200809L
    return strdup(string);
#else
    size_t buf_size = strlen(string) + 1;
    char *alloc;
    alloc = (char *)malloc(buf_size);
    if (alloc == nullptr)
    {
        fprintf(stderr, "Out of memory\n");
        fflush(stderr);
        exit(-1);
    }
#ifdef __STDC_LIB_EXT1__
    strcpy_s(alloc, buf_size, string);
#else
    strncpy(alloc, string, buf_size);
#endif
    alloc[buf_size - 1] = '\0';
    return alloc;
#endif
}
```

Then in general.h in the same directory, the following line gets added before line 29:

```C++
#define __STDC_WANT_LIB_EXT1__ 1
```

Reply:
The `strcpy_s` should work on Windows, and the `strdup` should work pretty much everywhere else. Theoretically, at least.

Reply:
Pushed that change in on the PR. Tested it as well, the vegasettings still works as expected.